internal/db/schema/migrations/postgres/10/04_session.up.sql

```diff
@@ -1,37 +1,38 @@
 begin;
 
-  create table session_credential_library (
+  create table session_credential_dynamic (
+    credential_id wt_public_id not null,
+    library_id wt_public_id not null,
+    constraint credential_dynamic_fkey
+      foreign key (credential_id, library_id)
+      references credential_dynamic (public_id, library_id)
+      on delete cascade
+      on update cascade,
     session_id wt_public_id not null
       constraint session_fkey
         references session (public_id)
         on delete cascade
         on update cascade,
-    credential_id wt_public_id not null
-      constraint credential_fkey
-        references credential (public_id)
-        on delete cascade
-        on update cascade,
-    credential_library_id wt_public_id not null
-      constraint credential_library_fkey
-        references credential_library (public_id)
-        on delete cascade
-        on update cascade,
-    target_credential_purpose text not null
+    credential_purpose text not null
       constraint target_credential_purpose_fkey
         references target_credential_purpose_enm (name)
         on delete restrict
         on update cascade,
     create_time wt_timestamp,
-    primary key(session_id, credential_id, credential_library_id, target_credential_purpose)
+    primary key(session_id, credential_id, library_id),
+    constraint session_credential_dynamic_library_id_credential_id_uq
+      unique(library_id, credential_id),
+    constraint session_credential_dynamic_credential_id_uq
```

Comment:
why not define this constraint below line 4 `credential_id wt_public_id not null,`

Reply:
I was grouping the unique constraints together.